src/syrupy/session.py

```diff
@@ -181,16 +197,3 @@ def _diff_snapshot_files(
 
     def _count_snapshots(self, snapshot_files: "SnapshotFiles") -> int:
         return sum(len(snaps) for snaps in snapshot_files.values())
-
-    def _in_snapshot_dir(self, path: str) -> bool:
-        parts = path.split(os.path.sep)
-        return SNAPSHOT_DIRNAME in parts
-
-    @lru_cache(maxsize=32)
-    def _walk_dir(self, root: str) -> Generator[str, None, None]:
```

Comment:
I know this is not new, but how does an LRU cache work with a generator here? Can you give a short example of a case where the cache provided values? Having trouble visualizing it for walk dir.

Reply:
I removed it in this PR. But an example would be we call this multiple times when discovering snapshots in the same test file using different assertion fixtures, the general idea is we would not need to walk the entire `__snapshot__` directory each time if we memoize it.

Reply:
Discussed offline `lru_cache` does not support generators, removed in this PR